editor/src/core/shared/atom-with-pub-sub.tsx

```diff
@@ -1,10 +1,19 @@
 import * as PubSub from 'pubsub-js'
 import * as React from 'react'
+import { unstable_batchedUpdates } from 'react-dom'
 import { useForceUpdate } from '../../components/editor/hook-utils'
 
+// From https://github.com/dai-shi/use-context-selector/blob/2dd334d727fc3b4cbadf7876b6ce64e0c633fd25/src/index.ts#L25
+const isSSR =
+  typeof window === 'undefined' ||
+  /ServerSideRendering/.test(window.navigator && window.navigator.userAgent)
+
+export const useIsomorphicLayoutEffect = isSSR ? React.useEffect : React.useLayoutEffect
```

Comment:
`Isomorphic`? I aint letting that one get past me!

Reply:
sorry Sean. you gotta let this go, it is _the_ common way to call react code that renders both on the backend and frontend isomorphic.